core/web/log_controller.go

```diff
@@ -0,0 +1,75 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+	"go.uber.org/zap/zapcore"
+)
+
+// LogController manages the logger config
+type LogController struct {
+	App chainlink.Application
+}
+
+type LogPatchRequest struct {
+	Level      string `json:"level"`
+	SqlEnabled *bool  `json:"sqlEnabled"`
+}
+
+// Patch sets a log level and enables sql logging for the logger
+func (cc *LogController) Patch(c *gin.Context) {
+	request := &LogPatchRequest{}
+	if err := c.ShouldBindJSON(request); err != nil {
+		jsonAPIError(c, http.StatusUnprocessableEntity, err)
+		return
+	}
+
+	if request.Level == "" && request.SqlEnabled == nil {
+		jsonAPIError(c, http.StatusInternalServerError, fmt.Errorf("please set either logLevel or logSql as params in order to set the log level"))
```

Comment:
http.StatusInternalServerError isn't the right code for this - we want to throw a 400 `http.StatusBadRequest`